### PR TITLE
Gas price component

### PR DESCRIPTION
**Pull request by 0xjocke**
more hooks

### CODE REVIEW

File: v2/components/MainActionCards/MainActionCards.tsx [hunk 1]
```diff
@@ -255,19 +257,35 @@ export const MainActionCardsUi: React.FC<UiProps> = (props) => {
...
+          Synthetix.burnSynthsToTarget({ ...gasOptions });
+        }}
+      >
+        Burn Max
+      </Button>
+      <EthGasPriceEstimator
+        onGasOptionChange={(x) => setGasOption(x)}
+        getGasLimit={() => Synthetix.estimateGas.burnSynthsToTarget()}
+        populateTransaction={() => Synthetix.populateTransaction.burnSynthsToTarget()}
+      />
```
Comment (bachstatter) [inline v2/components/MainActionCards/MainActionCards.tsx:280]: Example usage, will be removed

File: v2/lib/useGasPrice/useGasPrice.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          }
+        }
+        // When Testnet, Optimism network or missing baseFeePerGas we get the Gas Price through the provider
+        return getGasPriceFromProvider(provider);
+      } catch (e) {
+        throw new Error(`Could not fetch and compute network fee. ${e}`);
+      }
+    },
+    {
+      enabled: Boolean(networkId),
```
Comment (bachstatter) [inline v2/lib/useGasPrice/useGasPrice.ts:66]: We might want a refetch interval here

File: v2/components/EthGasPriceEstimator/EthGasPriceEstimator.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+};
+
+export const EthGasPriceEstimator: React.FC<{
+  gasLimit?: BigNumber;
+  gasPrices?: GasPrices;
+  optimismLayerOneFees?: Wei;
+}> = ({ gasLimit, gasPrices, optimismLayerOneFees }) => {
+  const { t } = useTranslation();
+
+  const [gasSpeed, _setGasSpeed /*TODO Will be used when we have a UI for picking speed*/] =
```
Comment (bachstatter) [inline v2/components/EthGasPriceEstimator/EthGasPriceEstimator.tsx:55]: Will fix this in the next PR, this one is big enough

File: v2/components/EthGasPriceEstimator/EthGasPriceEstimator.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    useState<'average' | 'fast' | 'fastest'>('average');
+
+  if (!gasLimit || !gasPrices) {
+    return <Text>Skeleton</Text>;
+  }
+  const gasPrice = gasPrices[gasSpeed];
+  const ethPriceRate = wei(1500); // todo
```
Comment (bachstatter) [inline v2/components/EthGasPriceEstimator/EthGasPriceEstimator.tsx:62]: Will fix this in the next PR, this one is big enough

File: v2/lib/useOptimismLayer1Fee/useOptimismLayer1Fee.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+  return wei(await OptimismGasPriceOracleContract.getL1Fee(serializedTxn));
+};
+export const useOptimismLayer1Fee = ({
+  populateTransaction,
+}: {
+  populateTransaction?: () => Promise<PopulatedTransaction>;
+}) => {
+  const { networkId } = useContext(ContractContext);
+  const serializedTransactionQuery = useQuery(
```
Comment (bachstatter) [inline v2/lib/useOptimismLayer1Fee/useOptimismLayer1Fee.ts:33]: Move the useQuery to the useQuery returned

**Review by noisekit — APPROVED** [review]

**Review by noisekit — APPROVED** [review]

**Review by jmzwar — APPROVED** [review]

File: v2/components/MainActionCards/MainActionCards.tsx [hunk 6]
```diff
@@ -258,16 +261,45 @@ export const MainActionCards: React.FC = () => {
   const { data: debtData } = useDebtData();
   const { data: feePoolData } = useFeePoolData();
   const { data: rewardsData } = useRewardsAvailable();
-  if (!debtData || !feePoolData || !rewardsData) return <p>Skeleton</p>;
+  const { data: Synthetix } = useSynthetix();
+
+  const getGasLimit = Synthetix?.signer
+    ? () => Synthetix.estimateGas.burnSynthsToTarget()
+    : undefined;
```
[the remark this answers is not in the document]
Comment (bachstatter) [reply v2/components/MainActionCards/MainActionCards.tsx:268]: Yea we might come up with something once we start using it more.. Couldn't see a nice pattern for it right now

**Pull request merged**